src/backend/executor/execCurrent.c

```diff
@@ -217,14 +223,34 @@ getCurrentOf(CurrentOfExpr *cexpr,
 		relispartition ||
 		get_rel_persistence(table_oid) == RELPERSISTENCE_TEMP)
 	{
+		bool matchInhertitanceTable;
+		Index i;
+
+		matchInhertitanceTable = false;
+		if (Gp_role == GP_ROLE_UTILITY && queryDesc->estate->es_rowmarks != NULL)
+		{
+			for (i = 0; i < queryDesc->estate->es_range_table_size; i++)
+			{
+				ExecRowMark *thiserm = queryDesc->estate->es_rowmarks[i];
+				if (thiserm == NULL ||
+					!RowMarkRequiresRowShareLock(thiserm->markType))
+					continue;		/* ignore non-FOR UPDATE/SHARE items */
+
+				if (thiserm->relid == table_oid)
+				{
+					matchInhertitanceTable = true;
+				}
+			}
+		}
+
 		/*
 		 * The target relation must directly match the cursor's relation. This throws out
 		 * the simple case in which a cursor is declared against table X and the update is
-		 * issued against Y. Moreover, this disallows some subtler inheritance cases where
-		 * Y inherits from X. While such cases could be implemented, it seems wiser to
-		 * simply error out cleanly.
```

Comment:
According to the comments of this function :
>  GPDB calls it before dispatching to make QEs get the same current position
>  of the cursor.

Do we need to change here? we won't dispatch in singlenode. Enable this in utility is another topic and not sure for its benefit or risk.

Reply:
test `portal` in `regress` suite tests inheritance cases at https://github.com/cloudberrydb/cloudberrydb/blob/10538adb70ac5e81abbe0cb16a9e1cf66ad808ec/src/test/regress/sql/portals.sql#L392. we need this to pass the test. or we have to disable this part of it.